### PR TITLE
update to latest 1.8.x k8s version for GKE (prow) and minikube (circleci)

### PR DESCRIPTION
**Pull request by ayj**
GKE 1.8.x is required to enable webhook config validation in e2e
tests. Update minikube k8s version at the same time to minimize
version skew.

### CODE REVIEW

**Comment by ayj** [comment]
fyi @jmuk for mixer validation webhooks

**Comment by rshriram** [comment]
Thanks a lot for upgrading circle ci as well!

File: prow/cluster_lib.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ ZONE=us-central1-f
 MACHINE_TYPE=n1-standard-4
 NUM_NODES=1
 CLUSTER_NAME=
-CLUSTER_VERSION=$(gcloud container get-server-config --project="${PROJECT_NAME}" --zone="${ZONE}" --format='value(defaultClusterVersion)')
+CLUSTER_VERSION="1.8.3-gke.0"
```
Comment (sebastienvas) [inline prow/cluster_lib.sh]: 1.8.4-gke.0 is available now. The problem with setting a fixed version is that gke does not keep them. What about something like:
IFS=';' VERSIONS=($(gcloud container get-server-config --project=istio-testing --zone=us-central1-f --format='value(validMasterVersions)'))
CLUSTER_VERSION="${VERSIONS[1]}"

Comment (ayj) [reply prow/cluster_lib.sh]: That seems works - updated.

**Comment by ayj** [comment]
re: `ci/circleci: pilot-integration-auth` failures (see quote below), I verified that minikube with k8s=1.8.0 works in my local environment with pilot/bin/e2e.sh. There failure seems to be specific to CircleCI

> serviceaccount "istio-pilot-service-account" created
Error from server (Timeout): error when creating "STDIN": Timeout: request did not complete within allowed duration
E1130 19:10:36.668369    9117 driver.go:181]

**Comment by rshriram** [comment]
@ayj were you able to reproduce circleci's setup (not minikube, its localkube with --vm-driver=none), and it still works? Looking at the logs, I see that its warning there is a new version of minikube in circleci. Does that help?

**Pull request merged**